Add encryption to in-cluster pod traffic (#911)
kinvolk/lokomotive: docs/configuration-reference/platforms/aws.md

  enable_tls_bootstrap = true

>   encrypt_pod_traffic = true

Reviewer: I think docs commit can go together with the implementation.

>   encrypt_pod_traffic = true

Reviewer: Docs can be committed with the implementation.

  disk_iops = var.disk_iops

>   network_mtu = 1480

Reviewer: I think this commit could be merged with the one changing the code.

| `region`                      | AWS region to use for deploying the cluster.                                                                                                                                               | "eu-central-1"  |    string    |  false   |
| `enable_aggregation`          | Enable the Kubernetes Aggregation Layer.                                                                                                                                                   |      true       |     bool     |  false   |
| `enable_tls_bootstrap`        | Enable TLS bootstraping for Kubelet.                                                                                                                                                       |      true       |     bool     |  false   |

> | `encrypt_pod_traffic`         | Enable in-cluster pod traffic.                                                                                                                                                             |      false      |     bool     |  false   |

Reviewer 2: ```suggestion
| `encrypt_pod_traffic`         | Enable in-cluster pod traffic. If true you need to reduce the default `network_mtu`, see its documentation.                                                                                                                                                             |      false      |     bool     |  false   |
```

> | `encrypt_pod_traffic`         | Enable in-cluster pod traffic. If true you need to reduce the default `network_mtu`, see its documentation.                                                                                |      false      |     bool     |  false   |

Reviewer 2: This doesn't apply anymore. Also, the first sentence doesn't make sense. How about this?

```suggestion
| `encrypt_pod_traffic`         | Enable in-cluster pod traffic encryption. If true `network_mtu` is reduced by 40 to make room for the encryption header.                                                                                |      false      |     bool     |  false   |
```

Please change all the other platforms as well.

| `disk_size`                   | Size of the EBS volume in GB.                                                                                                                                                              |       40        |    number    |  false   |
| `disk_type`                   | Type of the EBS volume (e.g. standard, gp2, io1).                                                                                                                                          |      "gp2"      |    string    |  false   |
| `disk_iops`                   | IOPS of the EBS volume (e.g 100).                                                                                                                                                          |        0        |    number    |  false   |

> | `network_mtu`                 | CNI interface MTU. Use 1440 if `encrypt_pod_traffic = true`. Use 8981 if using instances types with Jumbo frames and `encrypt_pod_traffic = false` else use 8941.                          |      1480       |    number    |  false   |

Reviewer 2: ```suggestion
| `network_mtu`                 | CNI interface MTU. Use 1440 if `encrypt_pod_traffic = true`. When using instance types with Jumbo frames, use 8981 (or 8941 if `encrypt_pod_traffic = true`).                          |      1480       |    number    |  false   |
```

Reviewer: I thought we could subtract it automatically, so they user don't run into troubles if they don't set it up?

Reviewer 2: That would be ideal but I'd be ok with having the user set this manually for now.

Reviewer 2: If it's easy to do we should just do it of course.

Reviewer: @knrt10 can you try implementing it? I think we could put it at Terraform template level, something like:
```
{{ if .Config.EncruptPodTraffic }}
network_mtu = {{ .Config.NetworkMTU }} - 40 # Wireguard overhead
{{- else }}
network_mtu = {{ .Config.NetworkMTU }}
{{- end }}
```

Author: Done, but it had to be implemented little differently, because we were setting default value in terraform, instead of that did that in go code, and it works

> | `network_mtu`                 | CNI interface MTU. Use 1440 if `encrypt_pod_traffic = true`. When using instance types with Jumbo frames, use 8981 (or 8941 if `encrypt_pod_traffic = true`).                              |      1480       |    number    |  false   |

Reviewer 2: This doesn't apply anymore. How about this?

```suggestion
| `network_mtu`                 | CNI interface MTU. When using instance types with Jumbo frames, use 8981.                              |      1480       |    number    |  false   |
```

> | `network_mtu`                 | CNI interface MTU. When using instance types with Jumbo frames, use 8981.                                                                                                                  |      1480       |    number    |  false   |

Reviewer: This change seems unrelated to the adding of encryption support.

Author: It was suggested by Iago or Johannes for fixing the sentences I guess. Should I add it in a separate commit?

Reviewer: Yes, please.
